spark2/src/test/java/org/apache/iceberg/examples/SchemaEvolutionTest.java

```diff
@@ -84,29 +98,56 @@ public void before() throws IOException {
 
   @Test
   public void addColumnToSchema() {
-    table.updateSchema().addColumn("publisher", Types.StringType.get()).commit();
 
-    Dataset<Row> df2 = spark.read().json(dataLocation + "new-books.json");
+    String fieldName = "publisher";
+    Schema schema = table.schema();
+    Assert.assertNull(schema.findField(fieldName));
+
+    table.updateSchema().addColumn(fieldName, Types.StringType.get()).commit();
+
+    Dataset<Row> df1 = spark.read()
+        .json(dataLocation + "books.json");
+    df1 = df1.withColumn(fieldName, new Column(Literal$.MODULE$.apply(null)))
+        .selectExpr("title", "price", "author", "cast(published as timestamp)", "genre", "publisher");
 
+    Dataset<Row> df2 = spark.read().json(dataLocation + "new-books.json")
+        .selectExpr("title", "price", "author", "cast(published as timestamp)", "genre", "publisher");
+    List<Row> expected = df1.union(df2)
+        .collectAsList();
+
+    // Append data
     df2.select(df2.col("title"), df2.col("price").cast(DataTypes.IntegerType),
         df2.col("author"), df2.col("published").cast(DataTypes.TimestampType),
         df2.col("genre"), df2.col("publisher")).write()
         .format("iceberg")
         .mode("append")
         .save(tableLocation.toString());
+
+    // Read iceberg table
+    Dataset<Row> iceberg = spark.read()
+        .format("iceberg")
+        .load(tableLocation.toString());
+
+    String error = QueryTest$.MODULE$.checkAnswer(iceberg, expected);
```

Comment:
This shouldn't use Spark test internals. There are assertions in Iceberg to check rows.